cmd/kubelet/app/options/options.go

```diff
@@ -394,6 +391,10 @@ func (f *KubeletFlags) AddFlags(mainfs *pflag.FlagSet) {
 	fs.MarkDeprecated("chaos-chance", "will be removed in a future version.")
 	fs.StringVar(&f.SeccompProfileRoot, "seccomp-profile-root", f.SeccompProfileRoot, "<Warning: Alpha feature> Directory path for seccomp profiles.")
 	fs.MarkDeprecated("seccomp-profile-root", "will be removed in 1.23, in favor of using the `<root-dir>/seccomp` directory")
+	fs.StringVar(&f.CloudProvider, "cloud-provider", f.CloudProvider, "The provider for cloud services. Set to `external` string for running with no cloud provider. If set, the cloud provider determines the name of the node (consult cloud provider documentation to determine if and how the hostname is used).")
```

Comment:
> Set to `external` string for running with no cloud provider

really? that's pretty weird, and doesn't match what we had before

Reply:
This is the preferred way to set the flag, since empty will raise warnings.

Reply:
> empty will raise warnings

that seems very strange to me... to run a standalone apiserver with no cloudprovider integration, I have to specify `--cloud-provider=external`?

Reply:
looking at the code, it just logs an info line saying no cloud provider was specified... that doesn't seem worth this confusion

Reply:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/server.go#L522

This flag return will be less verbose when specified `external`. A few options is to bring the name == "" to this check, or make external the default value instead.

Reply:
> Setting it to "" will give you a few warnings saying the cloud provider was not set, in case it was something like a typo on the flag name.

Typoed flags will produce errors that prevent the kubelet from starting. Typoed config fields names will log a warning. I think we should omit logging cloud-provider related lines when no cloud provider was configured.

Reply:
> Typoed flags will produce errors that prevent the kubelet from starting. Typoed config fields names will log a warning. 

That's a true statement, invalid values of flags will not allow starting it either.

> I think we should omit logging cloud-provider related lines when no cloud provider was configured.

I'm removing the logs related with the process in the PR.

Reply:
This looks better to me. sig-node has final lgtm